libs/framework/src/framework.c

```diff
@@ -237,6 +241,11 @@ celix_status_t framework_create(framework_pt *out, celix_properties_t* config) {
     celixThreadMutex_create(&framework->dispatcher.mutex, NULL);
     celixThreadMutex_create(&framework->frameworkListenersLock, NULL);
     celixThreadMutex_create(&framework->bundleListenerLock, NULL);
+    celix_thread_mutexattr_t muAttr;
+    celixThreadMutexAttr_create(&muAttr);
+    celixThreadMutexAttr_settype(&muAttr, CELIX_THREAD_MUTEX_RECURSIVE);
```

Comment:
I use it to combine `celix_framework_uninstallBundleEntry` and `celix_framework_installBundleInternal` into one operation. It can be fixed by some code refactoring.